test/unit/zendesk-uploader/client-wrapper.js

```diff
@@ -146,13 +146,39 @@ describe('ZendeskClientWrapper', () => {
             });
         });
 
-        it('should reject the promise if the request fails after 5 retries', () => {
+        it('should retry after twice as long as the last retry for the request when there are subsequent 500 errors', () => {
+            const stub = sandbox.stub(this.zendeskStub.articles, 'create').yields({statusCode: 503});
+            stub.onCall(3).yields(null, null, {});
+
+            let clock = sandbox.useFakeTimers();
+            let createArticle = this.zendeskClient.articles.create(123, {});
+
+            clock.tick(500 + 1000 + 2000);
+            return createArticle.then(() => {
+                expect(stub.callCount).to.be.equal(4);
+            });
+        });
+
+        it('should not wait longer than 8000ms on subsequent retries due to 500 errors', () => {
+            const stub = sandbox.stub(this.zendeskStub.articles, 'create').yields({statusCode: 503});
+            stub.onCall(6).yields(null, null, {});
+
+            let clock = sandbox.useFakeTimers();
+            let createArticle = this.zendeskClient.articles.create(123, {});
+
+            clock.tick(500 + 1000 + 2000 + 4000 + 8000 + 8000);
+            return createArticle.then(() => {
+                expect(stub.callCount).to.be.equal(7);
```

Comment:
I'm not sure this is checking what we need. All 7 calls to API may be done in first 500 ms for instance.